### DISCUSSION TITLE
How can I bypass a package's `exports` configuration and import arbitrary files?

### DISCUSSION THREAD

**Discussion by dstaley** [webpack/webpack#15973] [Q&A]
I'm using a third-party package that doesn't expose all its internal files through the `exports` field on `package.json`. I'd like to import an internal file from this package, but webpack (correctly!) respects the defined exports in the `exports` field. I know that I can completely disable support for the `exports` field by setting `config.resolve.exportsFields` to an empty array, but how can I either disable `exports` for this singular third-party dependency or bypass the `exports` config for a singular file import from said dependency?

For example, let's say the third-party package `super-rad` has the following in `package.json`:

```json
{
  "exports": {
    ".": "./main.js",
    "./sub/path": "./secondary.js",
    "./prefix/": "./directory/",
    "./prefix/deep/": "./other-directory/",
    "./other-prefix/*": "./yet-another/*/*.js"
  }
}
```

If I want to use `import thing from 'super-rad/internal/constants'`, how would I do that without completely disabling support for `exports` fields in other packages?

**Comment by vankop** [comment]
you can set `resolve.exportsFields = []` on `module.rules` level only for this dependency
